zebra-state/src/service/non_finalized_state.rs

```diff
@@ -97,19 +99,31 @@ impl NonFinalizedState {
             })
             .expect("commit_block is only called with blocks that are ready to be commited");
 
+        check::block_is_contextually_valid_for_chain(
+            &prepared,
+            self.network,
+            &parent_chain.mmr_hash(),
+        )?;
         parent_chain.push(prepared);
         self.chain_set.insert(parent_chain);
         self.update_metrics_for_committed_block(height, hash);
+        Ok(())
     }
 
     /// Commit block to the non-finalized state as a new chain where its parent
     /// is the finalized tip.
-    pub fn commit_new_chain(&mut self, prepared: PreparedBlock) {
+    pub fn commit_new_chain(
+        &mut self,
+        prepared: PreparedBlock,
+    ) -> Result<(), ValidateContextError> {
+        // TODO: will need to pass the MMR tree of the finalized tip to the newly built chain. How?
```

Comment:
The state service can get the history tree from the finalized state, and pass it to `commit_new_chain`.

Then we'll need to initialize the chain with the history tree:
1. Delete `impl Default for Chain`
2. Add a `Chain::new` method that takes a history tree

@dconnolly I expect we'll need a similar design to get the anchor data from the finalized state to the non-finalized state.

Reply:
Done